### PR TITLE
Update config-proxy.md

### PR DESCRIPTION
**Pull request by Goodiec**
This PR improves the proxy configuration doc and is part of [issue 41](https://github.com/jupyterhub/outreachy/issues/41)

### CODE REVIEW

File: docs/source/reference/config-proxy.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ server {
     listen 80;
     server_name HUB.DOMAIN.TLD;
 
-    # Tell all requests to port 80 to be 302 redirected to HTTPS
+    # Send all requests to port 80 to 302 and redirect to HTTPS
```
Comment (ryanlovett) [inline docs/source/reference/config-proxy.md:54]: This configuration line actually returns HTTP code 302 -- it does not redirect traffic from port 80 to port 302.

File: docs/source/reference/config-proxy.md [hunk 2]
```diff
@@ -101,18 +101,18 @@ server {
...
+server blocks as above for `NO_HUB` and simply add a line for the root directory
 of the site as well as the applicable location call:
 
 ```bash
 server {
     listen 80;
     server_name NO_HUB.DOMAIN.TLD;
 
-    # Tell all requests to port 80 to be 302 redirected to HTTPS
+    # Send all requests to port 80 to 302 and redirect to HTTPS
```
Comment (ryanlovett) [inline docs/source/reference/config-proxy.md:115]: This configuration line actually returns HTTP code 302 -- it does not redirect traffic from port 80 to port 302.

Comment (Goodiec) [reply docs/source/reference/config-proxy.md:115]: Thank you @ryanlovett, what should be the accurate comment there?

Comment (ryanlovett) [reply docs/source/reference/config-proxy.md:115]: There is some information about nginx's `return` directive at https://www.nginx.com/blog/creating-nginx-rewrite-rules/ and about the `302` HTTP code at https://en.wikipedia.org/wiki/HTTP_302. Here, (and above, in the other instance) nginx is redirecting browsers from the current non-HTTPS port 80 to the HTTPS url.

Comment (Goodiec) [reply docs/source/reference/config-proxy.md]: @ryanlovett, made changes, kindly review.

File: docs/source/reference/config-proxy.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ server {
     listen 80;
     server_name HUB.DOMAIN.TLD;
 
-    # Tell all requests to port 80 to be 302 redirected to HTTPS
+    # Send a redirect request using the $request_uri that was requested but via HTTPS
```
[the remark this answers is not in the document]
Comment (Goodiec) [reply docs/source/reference/config-proxy.md]: Thank you @ryanlovett!

**Comment by minrk** [comment]
Thanks @Goodiec and thanks @ryanlovett for the review!

**Pull request merged**